src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -574,9 +575,11 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
 
                 try
                 {
-                    SetData(await PrepareData(data, cancellationToken).ConfigureAwait(false));
-                    _applicationSettings = data;
+                    Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
+                    SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+                    watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);

```

Comment:
Wondering what the motivation was to put this here instead of right after `LoadKeyValuesRegisteredForRefresh`?

Reply:
It can be inside of the try statement and might fit better with the pattern of calling all three in the same spot, willing to change that. I just saw that it didn't need to be inside of the ExecuteWithFailOverPolicyAsync call so I looked further down to where watchedsettings was used.

Reply:
It is fine to be here, but if it stays here i'd recommend doing `_watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);`

My initial expectation was it to be right after the try

Reply:
We call it in `ExecuteWithFailOverPolicyAsync` in the other call location. Perhaps it is a good idea to just move it in there, right after `LoadKeyValuesRegisteredForRefresh`

Reply:
Oh that's true, makes sense thanks